### PR TITLE
Take a stab at specifying observer scope.

### PR DESCRIPTION
**Pull request by szager-chromium**
Issue #74.

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -138,8 +147,29 @@ intersection with the <a>intersection root</a>, as per the
 The IntersectionObserver interface</h3>
 
 The {{IntersectionObserver}} interface can be used to observe changes in the
-intersection of a <a>target</a> {{Element}} and a {{IntersectionObserver/root}}
-{{Element}} (or the top-level document's viewport).
+intersection of a {{IntersectionObserver/root}} {{Element}} (or the
+<a>top-level browsing context</a>'s visual viewport), and one or more <a>target</a> {{Elements}}.
```
Comment (mpb) [inline index.bs]: Elements point to https://dom.spec.whatwg.org/#elements which is an array-like collection.

Comment (szager-chromium) [inline index.bs]: Changed this to {{Element}}s, here and elsewhere

**Comment by mpb** [comment]
LGTM

**Comment by szager-chromium** [comment]
Ojan, any thoughts on this?

File: index.bs [hunk 2]
```diff
@@ -138,8 +147,29 @@ intersection with the <a>intersection root</a>, as per the
...
 The {{IntersectionObserver}} interface can be used to observe changes in the
-intersection of a <a>target</a> {{Element}} and a {{IntersectionObserver/root}}
-{{Element}} (or the top-level document's viewport).
+intersection of a {{IntersectionObserver/root}} {{Element}} (or the
+<a>top-level browsing context</a>'s visual viewport), and one or more <a>target</a> {{Element}}s.
+
+An {{IntersectionObserver}} with a {{IntersectionObserver/root}} {{Element}} can
+observe any <a>target</a> {{Element}} that is a descendant of the
+{{IntersectionObserver/root}} in the <a>node tree</a>, including {{Element}}s inside
+an <a>iframe</a> that belongs to the same <a>unit of related similar-origin browsing contexts</a>
```
Comment (ojanvafai) [inline index.bs]: As per recent issue discussions, this text isn't right anymore, right? Namely:
1. Elements inside similar-origin browsing context frames are no longer allowed.
2. Root needs to be an ancestor of target in the containingBlock chain.

File: index.bs [hunk 3]
```diff
@@ -138,8 +147,29 @@ intersection with the <a>intersection root</a>, as per the
...
+     <pre highlight="html">
+     &lt;div id="root"&gt;
+       &lt;iframe srcdoc="&lt;div id='target'&gt;"&gt;&lt;iframe&gt;
+     &lt;div&gt;
+     </pre>
+</div>
+
+An {{IntersectionObserver}} with no {{IntersectionObserver/root}} {{Element}}
+will automatically observe intersections with the <a>top-level browsing context</a>'s
+visual viewport, and valid <a>targets</a> include any {{Element}} in the
```
Comment (ojanvafai) [inline index.bs]: I'm not sold that we should be changing from layout to visual viewport. I think we should probably VC with me, you and @esprehn to hash it out.

File: index.bs (hunk 2 above)
Comment (szager-chromium) [inline index.bs]: Fixed.

**Comment by szager-chromium** [comment]
Added a definition for "implicit root", and used that term throughout.

PTAL

File: index.bs [hunk 4]
```diff
@@ -70,32 +87,27 @@ however, including (but not limited to):
...
+browser architectures in the face of filters, webgl, and other features). In all
+of these scenarios the information is useful even when delivered at a slight
+delay and without perfect compositing-result data.
+
+The Intersersection Observer API addresses the above issues by giving developers
+a new method to asynchronously query the position of an element with respect to
+other elements or the global viewport. The asynchronous delivery eliminates the
+need for costly DOM and style queries, continuous polling, and use of custom
+plugins. By removing the need for these methods it allows applications to
+significantly reduce their CPU, GPU and energy costs.
```
Comment (tabatkins) [inline index.bs]: I can't tell what you did or didn't change here.  When I generally reformatted the spec I switched to [semantic linefeeds](http://rhodesmill.org/brandon/2012/one-sentence-per-line/), which I use in all of my specs and which make diffs better.

If you didn't actually change anything, can you revert this back to the formatting it had?

File: index.bs [hunk 5]
```diff
@@ -18,21 +20,35 @@ url: http://w3c.github.io/requestidlecallback/#dfn-list-of-idle-request-callback
...
+		type: dfn; text: nested browsing contexts
 	urlPrefix: webappapis.html;
 		type: dfn; text: report the exception
 		type: dfn; text: event loop
 	urlPrefix: infrastructure.html;
 		type: dfn; text: rules for parsing dimension values
 		url: #dfn-callback-this-value; type: dfn; text: callback this value
+	urlPrefix: embedded-content.html;
+		url: #the-iframe-element; type: dfn; text: iframe
 urlPrefix: https://heycam.github.io/webidl/
```
Comment (tabatkins) [inline index.bs]: You can remove this - Bikeshed already knows about the iframe anchor.  (Also, type:dfn is wrong - it's type:element.)

File: index.bs [hunk 6]
```diff
@@ -41,6 +57,7 @@ spec:dom-ls
 spec: css-values-3; type: dfn
 	text: absolute length
 	text: dimension
+spec: embedded-content; type: element; text: iframe
```
Comment (tabatkins) [inline index.bs]: There's no spec called "embedded-content".  Bikeshed only knows of one "iframe" reference, so you shouldn't need this line at all.  (If you didn't know, there's a markup shorthand for element refs - `<{iframe}>`.)

**Comment by tabatkins** [comment]
The formatting no longer matches the rest of the spec, but otherwise it looks reasonable.

**Comment by ojanvafai** [comment]
lgtm

File: index.bs (hunk 5 above)
Comment (szager-chromium) [inline index.bs]: Done

File: index.bs (hunk 6 above)
Comment (szager-chromium) [inline index.bs]: Done

**Pull request merged**